tests/MongoDB.Driver.Tests/UnifiedTestOperations/UnifiedAggregateOnCollectionOperation.cs

```diff
@@ -0,0 +1,109 @@
+        private readonly AggregateOptions _options;
+        private readonly PipelineDefinition<BsonDocument, BsonDocument> _pipeline;
+
+        public UnifiedAggregateOnCollectionOperation(
+            IMongoCollection<BsonDocument> collection,
+            PipelineDefinition<BsonDocument, BsonDocument> pipeline,
+            AggregateOptions options)
+        {
+            _collection = collection;
+            _pipeline = pipeline;
+            _options = options;
+        }
+
+        public OperationResult Execute(CancellationToken cancellationToken)
+        {
+            try
+            {
+                var cursor = _collection.Aggregate(_pipeline, _options, cancellationToken);
+                var result = cursor.ToList();
+
+                return OperationResult.FromResult(new BsonArray(result));
+            }
+            catch (Exception exception)
+            {
+                return OperationResult.FromException(exception);
+            }
+        }
+
+        public async Task<OperationResult> ExecuteAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                var cursor = await _collection.AggregateAsync(_pipeline, _options, cancellationToken);
+                var result = await cursor.ToListAsync();
+
+                return OperationResult.FromResult(new BsonArray(result));
+            }
+            catch (Exception exception)
+            {
+                return OperationResult.FromException(exception);
+            }
+        }
+    }
+
+    public class UnifiedAggregateOnCollectionOperationBuilder
+    {
+        private readonly EntityMap _entityMap;
+
+        public UnifiedAggregateOnCollectionOperationBuilder(EntityMap entityMap)
+        {
+            _entityMap = entityMap;
+        }
+
+        public IUnifiedEntityTestOperation Build(string targetCollectionId, BsonDocument arguments)
+        {
+            var collection = _entityMap.GetCollection(targetCollectionId);
+
+            AggregateOptions options = null;
+            PipelineDefinition<BsonDocument, BsonDocument> pipeline = null;
+
+            foreach (var argument in arguments)
+            {
+                switch (argument.Name)
+                {
+                    case "pipeline":
+                        var stages = argument.Value.AsBsonArray.Cast<BsonDocument>();
+                        pipeline = new BsonDocumentStagePipelineDefinition<BsonDocument, BsonDocument>(stages);
+                        break;
+                    default:
+                        throw new FormatException($"Invalid AggregateOperation argument name: '{argument.Name}'");
+                }
+            }
+
+            if (pipeline.Stages.Last().OperatorName == "$out")
+            {
+                return new UnifiedAggregateToCollectionOperation(collection, pipeline, options);
+            }
+
+            return new UnifiedAggregateOnCollectionOperation(collection, pipeline, options);
```

Comment:
I like using if/then/else when two mutually exclusive paths exist:

    if (pipeline.Stages.Last().OperatorName == "$out")
    {
        return new UnifiedAggregateToCollectionOperation(collection, pipeline, options);
    }
    else
    {
        return new UnifiedAggregateOnCollectionOperation(collection, pipeline, options);
    }

What do you think?